webui/react/src/hooks/useSettings.ts

```diff
@@ -109,22 +104,25 @@ export const validateSetting = (config: SettingsConfigProp, value: unknown): boo
   return validateBaseType(config.type.baseType, value);
 };
 
-export const getDefaultSettings = <T>(config: SettingsConfig, storage: Storage): T => {
-  return config.settings.reduce((acc, prop) => {
+export const getDefaultSettings = <T>(config: SettingsConfig, storage: V1UserWebSetting[]): T => {
+  const currentPathSettings =
+    storage.filter((setting) => setting.storagePath?.includes(config.storagePath)) || [];
+  return config.settings.reduce<GenericSettings>((acc, prop) => {
     let defaultValue = prop.defaultValue;
     if (prop.storageKey) {
-      defaultValue = storage.getWithDefault(prop.storageKey, defaultValue);
+      const setting = currentPathSettings.find((stg) => stg.key === prop.storageKey);
+      defaultValue = setting?.value ? JSON.parse(setting.value) : defaultValue;
     }
     acc[prop.key] = defaultValue;
     return acc;
-  }, {} as GenericSettings) as unknown as T;
+  }, {}) as unknown as T;
 };
 
 export const queryParamToType = (type: BaseType, param: string | null): Primitive | undefined => {
   if (param === null || param === undefined) return undefined;
   if (type === BaseType.Boolean) return param === 'true';
```

Comment:
All data are passed as string from db. I feel like we should store data as it is in db, then we can get rid of unnecessary type conversion code